storage/src/test/java/org/apache/kafka/storage/internals/checkpoint/PartitionMetadataFileTest.java

```diff
@@ -0,0 +1,98 @@
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.storage.internals.checkpoint;
+
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.errors.InconsistentTopicIdException;
+
+import org.apache.kafka.storage.internals.log.LogDirFailureChannel;
+import org.apache.kafka.test.TestUtils;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class PartitionMetadataFileTest  {
+    private final File dir = TestUtils.tempDirectory();
+
+    @Test
+    public void testSetRecordWithDifferentTopicId() {
+        File file = PartitionMetadataFile.newFile(dir);
+        PartitionMetadataFile partitionMetadataFile = new PartitionMetadataFile(file, null);
+        Uuid topicId = Uuid.randomUuid();
+        assertDoesNotThrow(() -> partitionMetadataFile.record(topicId));
+        Uuid differentTopicId = Uuid.randomUuid();
+        assertThrows(InconsistentTopicIdException.class, () -> partitionMetadataFile.record(differentTopicId));
+    }
+
+    @Test
+    public void testSetRecordWithSameTopicId() {
+        File file = PartitionMetadataFile.newFile(dir);
+        PartitionMetadataFile partitionMetadataFile = new PartitionMetadataFile(file, null);
+        Uuid topicId = Uuid.randomUuid();
+        assertDoesNotThrow(() -> partitionMetadataFile.record(topicId));
+        assertDoesNotThrow(() -> partitionMetadataFile.record(topicId));
+    }
+
+    @Test
+    public void testMaybeFlushWithTopicIdPresent() {
+        File file = PartitionMetadataFile.newFile(dir);
+        PartitionMetadataFile partitionMetadataFile = new PartitionMetadataFile(file, null);
+
+        Uuid topicId = Uuid.randomUuid();
+        assertDoesNotThrow(() -> partitionMetadataFile.record(topicId));
+        assertDoesNotThrow(partitionMetadataFile::maybeFlush);
+
+        assertDoesNotThrow(() -> {
+            List<String> lines = Files.readAllLines(file.toPath());
+            assertEquals(2, lines.size());
+            assertEquals("version: 0", lines.get(0));
```

Comment:
it would be nice to add comments about those contents are from `PartitionMetadata#encode`. otherwise, it would be hard to debug if this test is broken by something changes in the future.

Reply:
ok! thanks